src/modules/mojo.rs

```diff
@@ -0,0 +1,155 @@
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::mojo::MojoConfig;
+use crate::formatter::StringFormatter;
+
+/// Creates a module with the current Mojo version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("mojo");
+    let config = MojoConfig::try_load(module.config);
+
+    let is_mojo_project = context
+        .try_begin_scan()?
+        .set_files(&config.detect_files)
+        .set_extensions(&config.detect_extensions)
+        .set_folders(&config.detect_folders)
+        .is_match();
+
+    if !is_mojo_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|variable, _| match variable {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    let mojo_version_output = context.exec_cmd("mojo", &["--version"])?.stdout;
+                    let version_items = mojo_version_output.split(' ').collect::<Vec<&str>>();
+
+                    if version_items.len() <= 1 {
+                        return None;
+                    }
+
+                    if version_items.len() == 2 || !config.show_commit {
+                        return Some(Ok(version_items[1].trim().to_string()));
+                    }
```

Comment:
Instead of manual indexing, I would prefer something like this:
```
                    let (version, hash) = match version_items[..] {
                        [_, version] => (version.trim(), None),
                        [_, version, hash, ..] => (version, Some(hash)),
                        _ => {
                            log::debug!("Unexpected `mojo --version` output: {mojo_version_output}");
                            return None;
                        }
                    }
```